.travis.yml

```diff
@@ -1,6 +1,14 @@
 language: cpp
 sudo: required
 
+os:
+  - linux
+  - osx
+
+compiler:
+  - gcc
+  - clang
```

Comment:
@lucasdemarchi Actually builds don't timeout, but jobs do, so adding more targets inside the same job is what would actually be bad. But I think that it's possible to run two new jobs for clang.
Another question: on OS X you only want to run clang for SITL right?

@8W9aG This is important, @lucasdemarchi only wants clang to run for SITL, not for all targets. By the way, I'm not sure it's possible to compile with clang on Travis while using the stdlib from gcc-4.9 - you can see here a build I tried for SITL with clang: https://travis-ci.org/OXINARF/ardupilot/jobs/113230126#L7731  That error was only fixed in clang-3.4.2 (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=744792) and Ubuntu Precise (used in Travis) only has clang-3.4.1